src/eval.c

```diff
@@ -728,6 +731,21 @@ NULL
             addReplyError(c, "Use SCRIPT DEBUG YES/SYNC/NO");
             return;
         }
+    } else if (c->argc == 3 && !strcasecmp(c->argv[1]->ptr, "dump")) {
+        dictEntry *de;
+        luaScript *ls;
+
+        if (sdslen(c->argv[2]->ptr) != 40) {
+            addReplyError(c, "Sha1 length must be 40");
```

Comment:
do we want to return a noscripterr?

Reply:
I think it would be better to return an error, as users can intuitively know what the error is

Reply:
The -ERR vs null reply is a little inconsistent here.

We validate the length here, but we still accept invalid SHA1 strings. For strings with 40 characters other than [0-9a-f], we just return null.

Either we catch all invalid SHA1 values or we shouldn't try to catch them at all.

An alternative is that we do like EVALSHA, we return a -NOSCRIPT error for non-existing scripts and also for invalide SHA1 values. It's kind of the same error.

Reply:
> An alternative is that we do like EVALSHA, we return a -NOSCRIPT error for non-existing scripts and also for invalide SHA1 values. It's kind of the same error.

yean, i prefer to return noscript

Reply:
Got it